[Frontend] Refactor positions list to the applications

jacobljh opened the pull request:
Please see #128 and #145 for details

Refactor positions applied to applicants so that one application is associated with all the positions applied with preference level and the applicant's info.

dependabot-preview[bot] committed:
Build(deps): [Security] Bump tar from 2.2.1 to 2.2.2 in /frontend

Bumps [tar](https://github.com/npm/node-tar) from 2.2.1 to 2.2.2. **This update includes security fixes.**
- [Release notes](https://github.com/npm/node-tar/releases)
- [Commits](https://github.com/npm/node-tar/compare/v2.2.1...v2.2.2)

jacobljh committed:
fix(frontend) merge applicant positions into applicants

jacobljh committed:
fix(frontend) merge applicant positions into applicants

jacobljh committed:
fix(frontend) merge applicant positions into applicants

jacobljh committed:
fix(frontend) merge applicant positions into applicants

jacobljh committed:
refactor(frontend) move components from applicants_positions to applicants

jacobljh committed:
refactor(frontend) move components from applicants_positions to applicants

jacobljh committed:
refactor(frontend) move components from applicants_positions to applicants

jacobljh committed:
refactor(frontend) move components from applicants_positions to applicants

dependabot-preview[bot] committed:
Build(deps): [Security] Bump nokogiri from 1.10.1 to 1.10.3 in /api

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.10.1 to 1.10.3. **This update includes security fixes.**
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.10.1...v1.10.3)

Signed-off-by: dependabot[bot] <support@dependabot.com>

jacobljh committed:
refactor(frontend) fix routing error, name scope collisions, merge redux store

jacobljh committed:
refactor(frontend) fix routing error, name scope collisions, merge redux store

jacobljh committed:
Merge branch 'TAPP-128' of https://github.com/uoft-tapp/tapp into TAPP-128
rebase master

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -1,10 +1,14 @@
+// action used by applicant form view
 export const createNewApplication = payload => async dispatch => {
     const res = await fetch("/api/v1/applicants", {
         method: "POST",
```
We need to add the credential sending to this request. Once Shibboleth is activated, all requests sent without credentials will fail.

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -24,4 +28,17 @@ export const createNewApplication = payload => async dispatch => {
+export const fetchPositionsSuccess = payload => ({ type: FETCH_POSITIONS_SUCCESS, payload })
+export const fetchPositions = () => async dispatch => {
+    const res = await fetch("/api/v1/positions")
+    const data = await res.json()
```
This needs to be in a `try .. catch` block. `await` turns failed promises into thrown errors.

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -24,4 +28,17 @@ export const createNewApplication = payload => async dispatch => {
+// action used by applicant positions view
+export const fetchPositionsSuccess = payload => ({ type: FETCH_POSITIONS_SUCCESS, payload })
+export const fetchPositions = () => async dispatch => {
+    const res = await fetch("/api/v1/positions")
```
Credentials.

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -24,4 +28,17 @@ export const createNewApplication = payload => async dispatch => {
+    const res = await fetch("/api/v1/positions")
+    const data = await res.json()
+    if (res.status === 200) {
+        dispatch(fetchPositionsSuccess(data))
```
Wouldn't it be better to await the JSON in here after you know the fetch was a success?

siefkenj committed:
Added travis-ci testing

siefkenj committed:
Add tests folder

siefkenj committed:
Fixed typo

siefkenj committed:
Added steps to travis setup

siefkenj committed:
More Travis changes

siefkenj committed:
More Travis adjustments

siefkenj committed:
Added build matrix to Travis

siefkenj committed:
Travis change

siefkenj committed:
Change Travis

siefkenj committed:
Travis Changes

siefkenj committed:
Travis update

siefkenj committed:
Travis change

siefkenj committed:
Travis changes

siefkenj committed:
Allow formatting tests to fail

siefkenj committed:
Add eslint checks

siefkenj committed:
Added babel as the default parser for eslint

siefkenj committed:
Install eslint dependency

jacobljh committed:
refactor(frontend) add error handling to await keyword, refined conditioning logic

jacobljh committed:
refactor(frontend) add error handling to await keyword, refined conditioning logic

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -33,10 +33,18 @@ export const createNewApplication = payload => async dispatch => {
+            const data = await res.json()
+            dispatch(fetchPositionsSuccess(data))
+        } catch(err) {
+            alert(err);
```
Don't use `alert`. You should be dispatching any errors

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -33,10 +33,18 @@ export const createNewApplication = payload => async dispatch => {
 export const fetchPositions = () => async dispatch => {
-    const res = await fetch("/api/v1/positions")
-    const data = await res.json()
+    try{
```
Run your code through `prettier`. There should be a space before `{`

siefkenj commented on frontend/src/modules/application/actions.js:
```diff
@@ -33,10 +33,18 @@ export const createNewApplication = payload => async dispatch => {
-    const res = await fetch("/api/v1/positions")
-    const data = await res.json()
+    try{
+        var res = await fetch("/api/v1/positions")
```
We need to send credentials with every `fetch` otherwise it will fail when shibboleth is active.

siefkenj reviewed (commented):
See if you can remove `erd.svg` from your PR. It gets rebuilt automatically, and it will be annoying to have merge conflicts because of it...

michellemtchai committed:
fix(db): remove preference_levels table
removed preference_level table from schema
removed preference_level model
generate new erd diagram
resolve issue #150

michellemtchai committed:
feat(db): add nag_count & email_count to offers
added nag_count and email_count to offers table
updated erd diagram
resolved issue #147

michellemtchai committed:
fix(db): remove preference_level_id & add preference_level
removed preference_level from position_preference table and model
added preference_level column to position_preference
updated erd diagram

michellemtchai committed:
fix(db): add migration changes
added offer mode, schema, erd diagram after migration

reidka committed:
Merge pull request #99 from uoft-tapp/dependabot/bundler/api/nokogiri-1.10.3

Build(deps): [Security] Bump nokogiri from 1.10.1 to 1.10.3 in /api

reidka committed:
Merge pull request #115 from uoft-tapp/dependabot/npm_and_yarn/frontend/tar-2.2.2

Build(deps): [Security] Bump tar from 2.2.1 to 2.2.2 in /frontend

michellemtchai committed:
fix(db): remove email_count from offers
removed email_count from offers table/model
updated erd diagram

jacobljh committed:
refactor(frontend) prettier formatting (indent of 4 spaces)

jacobljh committed:
refactor(frontend) prettier formatting (indent of 4 spaces)

reidka committed:
Merge pull request #149 from siefkenj/travis

Added travis-ci testing

reidka committed:
Merge pull request #151 from michellemtchai/tapp-150

[Database] fix errors in schema

Commit:
enhancement(frontend): Update react-bootstrap & boostrap

This patch updates react-bootstrap & bootstrap to versions 1.0.0-beta9 & 4.3.1 respectively.  Obsolete react-bootstrap components are replaced.

siefkenj committed:
Merge pull request #156 from thezane/react-bootstrap

enhancement(frontend): Update react-bootstrap & boostrap

siefkenj committed:
Formatted all files with prettier

siefkenj committed:
Cleanup eslint errors

siefkenj committed:
Run formatting checks in travis

siefkenj committed:
Merge pull request #157 from siefkenj/prettier

[frontend] Format all code with prettier and eslint

siefkenj committed:
Added `apiGET` and `apiPOST` utility functions

siefkenj committed:
Fix eslint errors

siefkenj committed:
[travis] Return non-zero exit code when run-*-tests.sh scripts fail

siefkenj committed:
Test failure

siefkenj committed:
No more travis error :-)

siefkenj committed:
Merge pull request #159 from siefkenj/travis

[travis] Return non-zero exit code when `run-*-tests.sh` scripts fail

siefkenj committed:
Fixed typo

siefkenj committed:
Merge pull request #158 from siefkenj/seeddata

[frontend] Added `apiGET` and `apiPOST` utility functions

siefkenj committed:
Missing `await` keyword

siefkenj committed:
Merge pull request #162 from siefkenj/seeddata

[frontend] Missing `await` keyword

siefkenj committed:
[frontend] Upgraded package libraries

siefkenj committed:
Merge pull request #163 from siefkenj/seeddata

[frontend] Upgraded package libraries

siefkenj committed:
[frontend] Prevent page from reloading when nav link clicked

siefkenj committed:
Prettier

siefkenj committed:
Merge pull request #164 from siefkenj/seeddata

[frontend] Prevent page from reloading when nav link clicked

jacobljh committed:
refactor(frontend) use apiGET, apiPOST for fetching positions, posting applications

jacobljh committed:
refactor(frontend) use apiGET, apiPOST for fetching positions, posting applications

jacobljh committed:
refactor(frontend) rebase from master

jacobljh committed:
refactor(frontend) remove git confliction tags
